src/functions/pollUsers/domain/examiner-cache-reconciler.ts

```diff
@@ -1,9 +1,23 @@
-import { cacheStaffNumbers, uncacheStaffNumbers } from '../framework/repo/dynamodb/cached-examiner-repository';
+import {
+  uncacheStaffNumbers, cacheStaffDetails,
+} from '../framework/repo/dynamodb/cached-examiner-repository';
+import { StaffDetail } from '../../../common/application/models/staff-details';
 
-export const reconcileActiveAndCachedExaminers = async (activeStaffNumbers: string[], cachedStaffNumbers: string[]) => {
-  const staffNumbersToCache = activeStaffNumbers.filter(staffNumber => !cachedStaffNumbers.includes(staffNumber));
-  await cacheStaffNumbers(staffNumbersToCache);
+export const reconcileActiveAndCachedExaminers
+= async (activeStaffDetails: StaffDetail[], cachedStaffNumbers: string[]) => {
+
+  const staffDetailsToCache =
+  activeStaffDetails.filter(staffDetail => !cachedStaffNumbers.includes(staffDetail.staffNumber));
+  await cacheStaffDetails(staffDetailsToCache);
+
+  const staffNumbersToUncache = [];
+  cachedStaffNumbers.forEach((staffNumber) => {
```

Comment:
Couldn't you `map` the `StaffDetail` objects to their `staffNumber` and eliminate this nested loop?